docs/design/2020-09-28-cluster-read-only-mode.md

```diff
@@ -0,0 +1,233 @@
+
+#### Case 3, other users have held a write lock
+
+| server 1            | server 2                   |
+| ---                 | ---                        |
+| client 1            | client 2                   |
+|                     | lock table for write       |
+| turn on read_only   |                            |
+| block               |                            |
+| block               | unlock table               |
+| read_only turned on |                            |
+|                     | lock table for write       |
+|                     | error, read_only turned on |
+
+In this case, 2 users are involved, one has a table locked for write, and the other wants to turn on the read only mode, the attempt will be blocked until the write lock is released.
+
+
+#### Case 4, attempt blocked due to held write lock on t1, while other client update on t2
+
+MySQL's behavior:
+
+| server 1            | server 1            | server 1       |
+| ---                 | ---                 | ---            |
+| client 1            | client 2            | client 3       |
+|                     | lock table t1 write |                |
+| turn on read_only   |                     |                |
+| block               |                     | insert into t2 |
+| block               | unlock table        | block          |
+| read_only turned on |                     | insert ok      |
+
+TiDB's behavior (designed):
+
+| server 1            | server 2            | server 3         |
+| ---                 | ---                 | ---              |
+| client 1            | client 2            | client 3         |
+|                     | lock table t1 write |                  |
+| turn on read_only   |                     |                  |
+| block               |                     | insert into t2   |
+| block               | unlock table        | fail immediately |
+| read_only turned on |                     |                  |
+
+In this case, MySQL will also block the update, and once the table is unlocked, both update and the enabling attempt will succeed.
+
+However, it would be too difficult to achieve this in a distributed environment, therefore the proposed solution is to fail the update immediately, since the enabling attempt will eventually succeed, it doesn’t matter if we act like the update comes after the enabling of the `read_only` mode.
+
+In a sentence, attempts to turn on `read_only` will never fail, it will block until committing transactions and write locks held from/by unprivileged users before the attempt on all TiDB servers are committed or released, and TiDB server enters a `read_only` mode as long as it receives the update of the global variable, hence no blockage will be caused by turning on `read_only`, queries either fail immediately or get processed.
```

Comment:
"In a sentence, attempts to turn on `read_only` will never fail"

Would it be more clear/correct to say this instead:

"SET GLOBAL read_only=1 will always take priority over DML, and will never fail because of long running DML or DDL. The failure modes are only as a result of not being able to contact all TiDB servers."